modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/source/ExtensionMessageSource.java

```diff
@@ -382,4 +385,27 @@ private void notifyExceptionAndShutDown(Throwable exception)
         LOGGER.error(String.format("Message source '%s' on flow '%s' threw exception. Shutting down it forever...", source.getName(), flowConstruct.getName()), exception);
         shutdown();
     }
+
+    private Exception enrichException(Throwable t)
+    {
+
+        Optional<ExceptionEnricherFactory> exceptionEnricherFactory = sourceModel.getExceptionEnricherFactory();
+        if (!exceptionEnricherFactory.isPresent())
+        {
+            exceptionEnricherFactory = extensionModel.getExceptionEnricherFactory();
+        }
+
+        Exception e;
```

Comment:
reuse the code at org.mule.module.extension.internal.runtime.DefaultExecutionMediator#processException instead. A refactor might be needed to make that code shareable